crates/node-core/src/args/network.rs

```diff
@@ -227,31 +231,38 @@ pub struct DiscoveryArgs {
     #[arg(id = "discovery.port", long = "discovery.port", value_name = "DISCOVERY_PORT", default_value_t = DEFAULT_DISCOVERY_PORT)]
     pub port: u16,
 
-    /// The UDP address to use for devp2p peer discovery version 5.
-    #[arg(id = "discovery.v5.addr", long = "discovery.v5.addr", value_name = "DISCOVERY_V5_ADDR",
-    default_value_t = DEFAULT_DISCOVERY_V5_ADDR)]
-    pub discv5_addr: IpAddr,
+    /// The UDP IPv4 address to use for devp2p peer discovery version 5.
+    #[arg(id = "discovery.v5.addr.ipv4", long = "discovery.v5.addr.ipv4", value_name = "DISCOVERY_V5_ADDR_IPV4", default_value = None)]
+    pub discv5_addr_ipv4: Option<Ipv4Addr>,
+
+    /// The UDP IPv6 address to use for devp2p peer discovery version 5.
+    #[arg(id = "discovery.v5.addr.ipv6", long = "discovery.v5.addr.ipv6", value_name = "DISCOVERY_V5_ADDR_IPV6", default_value = None)]
+    pub discv5_addr_ipv6: Option<Ipv6Addr>,
 
-    /// The UDP port to use for devp2p peer discovery version 5.
-    #[arg(id = "discovery.v5.port", long = "discovery.v5.port", value_name = "DISCOVERY_V5_PORT",
+    /// The UDP IPv4 port to use for devp2p peer discovery version 5.
+    #[arg(id = "discovery.v5.port.ipv4", long = "discovery.v5.port.ipv4", value_name = "DISCOVERY_V5_PORT_IPV4",
```

Comment:
should we alias v4 or v5 to the older names?
otherwise this would break existing setups?